cartography/sync.py

```diff
@@ -93,43 +93,51 @@ def run_with_config(sync, config):
     neo4j_auth = None
     if config.neo4j_user or config.neo4j_password:
         neo4j_auth = (config.neo4j_user, config.neo4j_password)
-    try:
-        neo4j_driver = GraphDatabase.driver(
-            config.neo4j_uri,
-            auth=neo4j_auth,
-        )
-    except neobolt.exceptions.ServiceUnavailable as e:
-        logger.debug("Error occurred during Neo4j connect.", exc_info=True)
-        logger.error(
-            (
-                "Unable to connect to Neo4j using the provided URI '%s', an error occurred: '%s'. Make sure the Neo4j "
-                "server is running and accessible from your network."
-            ),
-            config.neo4j_uri,
-            e,
-        )
-        return
-    except neobolt.exceptions.AuthError as e:
-        logger.debug("Error occurred during Neo4j auth.", exc_info=True)
-        if not neo4j_auth:
-            logger.error(
-                (
-                    "Unable to auth to Neo4j, an error occurred: '%s'. cartography attempted to connect to Neo4j "
-                    "without any auth. Check your Neo4j server settings to see if auth is required and, if it is, "
-                    "provide cartography with a valid username and password."
-                ),
-                e,
+
```

Comment:
I don't think these changes are related to Docker support.
Would it be possible to extract them into their own PR?

Reply:
It is related.

Docker doesn't support waiting for containers to become healthy before spinning up the next container.

This retry mechanism allows docker to spin up cartography when neo4j is still booting up. Without this cartography will immediately fail and exit because neo4j won't have spun up yet.

Reply:
Can we do something like this instead?
https://docs.docker.com/compose/startup-order/

Reply:
Startup order does not wait for the container to be healthy. It only ensures order of initialization.

In the link you provided, it specifically says
```
The problem of waiting for a database (for example) to be ready is really just a subset of a much larger problem of distributed systems. In production, your database could become unavailable or move hosts at any time. Your application needs to be resilient to these types of failures.

To handle this, design your application to attempt to re-establish a connection to the database after a failure. If the application retries the connection, it can eventually connect to the database.
```